src/core/qgsproject.h

```diff
@@ -991,6 +1010,8 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
 
     std::unique_ptr<QgsMapThemeCollection> mMapThemeCollection;
 
+    std::unique_ptr<QgsLabelingEngineSettings> mLabelingEngineSettings;
```

Comment:
Like this we can just have a forward declaration of the class rather than having to #include its header